ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml

```diff
@@ -15,36 +14,53 @@
 
 - name: Reload Daemon
   shell: systemctl daemon-reload
-
-- name: Generate modules.dep and map files
-  shell: depmod -a
-  register: depmod_reg
+- name: Run cmd to create nbd 128 devices
+  shell: sudo modprobe nbd nbds_max=128
+  register: cmd_nbd
   ignore_errors: yes
 
 - debug:
-    msg: "{{ depmod_reg }}"
-  when: depmod_reg.rc != 0
+    msg: "{{ cmd_nbd }}"
+  when: cmd_nbd.changed == False
+  ignore_errors: yes
 
-- name: Create nbd 128 devices
-  shell: sudo modprobe nbd nbds_max=128
-  register: crt_nbd
+- name: Create nbd config file
+  lineinfile:
+    path: /etc/modprobe.d/nbd.conf
+    line: 'options nbd nbds_max=128'
+    create: yes
+  ignore_errors: yes
+  
+- name: Load nbd module into Linux Kernel
+  lineinfile:
+    path: /etc/modules-load.d/openstack-ansible.conf
+    line: 'nbd'
+    create: no
   ignore_errors: yes
 
+- name: Run cmd to create rbd
+  shell: sudo modprobe rbd
+  register: cmd_rbd
+  ignore_errors: yes
+ 
 - debug:
-    msg: "{{ crt_nbd }}"
-  when: crt_nbd.rc != 0
+    msg: "{{ cmd_rbd }}"
+  when: cmd_rbd.changed == False
+  ignore_errors: yes
 
-- name: Load nbd 128 devices
+- name: Create rbd config file
   lineinfile:
-    path: /etc/rc.modules
-    line: 'depmod -a && sudo modprobe nbd nbds_max=128'
+    path: /etc/modprobe.d/rbd.conf
+    line: 'options rbd'
```

Comment:
This is not needed. we are not using any options while loading rbd module.

Reply:
right

Reply:
The task Create rbd config file is creating **/etc/modprobe.d/rbd.conf** with data set  **options rbd**


The task is to create the rbd.conf file not to load into the kernel module